### PR TITLE
Add abstracted access to Device Attestation Credentials

### PR DESCRIPTION
**Pull request by tcarmelveilleux**
#### Problem

- PR #8848 introduced some abstraction of access, but in
  a way that was not dependency injectable, and that was tied
  to the ConfigurationManager.
- Issue #8606 identifies a possible way to make per-product factory data
  more easy to abstract.

#### Change overview

- This PR introduces a true injectable/replaceable abstraction
  for device attestation credentials access, and takes the example
  data and tests from #8848 and makes them available as bootstrap
  version

Issue #8435
Issue #8606

#### Testing

Testing done: added unit tests for entire feature, and added
example to build.

### CODE REVIEW

**Comment by woody-apple** [comment]
@pan-apple ?

File: src/credentials/DeviceAttestationCredsAccess.cpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    CHIP_ERROR SignWithDeviceAttestationKey(const ByteSpan & digest_to_sign, MutableByteSpan & out_signature_buffer) override
+    {
+        (void) digest_to_sign;
+        (void) out_signature_buffer;
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
+};
+
+// Default to avoid nullptr on getter and cleanly handle new products/clients before
+// their provide their own.
```
Comment (msandstedt) [inline src/credentials/DeviceAttestationCredsAccess.cpp:62]: Nit:

```suggestion
// they provide their own.
```

Comment (tcarmelveilleux) [reply src/credentials/DeviceAttestationCredsAccess.cpp]: Done

**Review by woody-apple — APPROVED** [review]

**Review by mspang — APPROVED** [review]
Looks good. I would start a bikeshed of the name though - maybe accessor -> provider ? "Access" seems like it could just as easily apply to the client of this interface which suggests it may not be the most descriptive name for the interface itself.

Also this PR seems to switch between local variable naming styles.

**Comment by andy31415** [comment]
@chrisdecenzo @Damian-Nordic @hawk248 @jepenven-silabs @msandstedt ?

**Review by msandstedt — APPROVED** [review]

**Comment by tcarmelveilleux** [comment]
> Looks good. I would start a bikeshed of the name though - maybe accessor -> provider ? "Access" seems like it could just as easily apply to the client of this interface which suggests it may not be the most descriptive name for the interface itself.
> 

Moved to `Provider` suffix, since it's better than `Accessor`.

> Also this PR seems to switch between local variable naming styles.

Fixed that part

**Review by mspang — APPROVED** [review]

**Pull request merged**